### PR TITLE
feat: add support to applications API

### LINKED ISSUE #200: Applications API support
Crowdin has recently added new endpoints for managing custom applications. Support for these endpoints is missing in this Crowdin API client.

We need to add support for the following endpoints:

- [Get Application Data](https://developer.crowdin.com/api/v2/#operation/api.applications.api.get)
- [Update or Restore Application Data](https://developer.crowdin.com/api/v2/#operation/api.applications.api.put)
- [Add Application Data](https://developer.crowdin.com/api/v2/#operation/api.applications.api.post)
- [Delete Application Data](https://developer.crowdin.com/api/v2/#operation/api.applications.api.delete)
- [Edit Application Data](https://developer.crowdin.com/api/v2/#operation/api.applications.api.patch)

### PR DESCRIPTION
**Pull request by sugan0tech**

### CODE REVIEW

**Comment by sugan0tech** [comment]
@andrii-bodnar made the changes, please provide the necessary changes.

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public ApplicationsApi(Credentials credentials) {
+        super(credentials);
+    }
+
+    public ApplicationsApi(Credentials credentials, ClientConfig clientConfig) {
+        super(credentials, clientConfig);
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application.
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:23]:
```suggestion
     * @param applicationIdentifier identifier of the application
```

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     * @param path The path is implemented by the application
+     * @return application data
+     * @param request request object
+     * @see <ul>
+     * <li><a href="https://developer.crowdin.com/api/v2/#operation/api.applications.api.put" target="_blank"><b>API Documentation</b></a></li>
+     * </ul>
+     */
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:45]:
```suggestion
    /**
     * @param applicationIdentifier identifier of the application
     * @param path The path is implemented by the application
     * @param request request object
     * @return application data
     * @see <ul>
     * <li><a href="https://developer.crowdin.com/api/v2/#operation/api.applications.api.put" target="_blank"><b>API Documentation</b></a></li>
<li><a href="https://developer.crowdin.com/enterprise/api/v2/#operation/api.applications.api.put" target="_blank"><b>Enterprise API Documentation</b></a></li>
     * </ul>
     */
```

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    public ResponseObject<ApplicationData> updateOrReplaceApplicationData(String applicationIdentifier, String path, UpdateOrReplaceApplicationDataRequest request) throws HttpException, HttpBadRequestException {
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:46]: `updateOrReplaceApplicationData` -> `updateOrRestoreApplicationData`

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        String builtUrl = String.format("%s/applications/%s/api/%s", this.url, applicationIdentifier, path);
+        ApplicationDataResponseObject response = this.httpClient.put(builtUrl, request, new HttpRequestConfig(), ApplicationDataResponseObject.class);
+        return ResponseObject.of(response.getData());
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application.
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:53]:
```suggestion
     * @param applicationIdentifier identifier of the application
```

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+     * @param path The path is implemented by the application
+     * @return application data
+     * @param request request object
+     * @see <ul>
+     * <li><a href="https://developer.crowdin.com/api/v2/#operation/api.applications.api.post" target="_blank"><b>API Documentation</b></a></li>
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:58]: add "Enterprise API Documentation" link

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        String builtUrl = String.format("%s/applications/%s/api/%s", this.url, applicationIdentifier, path);
+        ApplicationDataResponseObject response = this.httpClient.post(builtUrl, request, new HttpRequestConfig(), ApplicationDataResponseObject.class);
+        return ResponseObject.of(response.getData());
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application.
+     * @param path The path is implemented by the application
+     * @see <ul>
+     * <li><a href="https://developer.crowdin.com/enterprise/api/v2/#operation/api.projects.members.delete" target="_blank"><b>Enterprise API Documentation</b></a></li>
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:71]: Enterprise API Documentation

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+     * </ul>
+     */
+    public ResponseObject<ApplicationData> addApplicationData(String applicationIdentifier, String path, AddApplicationDataRequest request) throws HttpException, HttpBadRequestException {
+        String builtUrl = String.format("%s/applications/%s/api/%s", this.url, applicationIdentifier, path);
+        ApplicationDataResponseObject response = this.httpClient.post(builtUrl, request, new HttpRequestConfig(), ApplicationDataResponseObject.class);
+        return ResponseObject.of(response.getData());
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application.
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:68]:
```suggestion
     * @param applicationIdentifier identifier of the application
```

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        String builtUrl = String.format("%s/applications/%s/api/%s", this.url, applicationIdentifier, path);
+        ApplicationDataResponseObject response = this.httpClient.put(builtUrl, request, new HttpRequestConfig(), ApplicationDataResponseObject.class);
+        return ResponseObject.of(response.getData());
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application.
+     * @param path The path is implemented by the application
+     * @return application data
+     * @param request request object
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:56]:
```suggestion
     * @param request request object
     * @return application data
```

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        String builtUrl = String.format("%s/applications/%s/api/%s", this.url, applicationIdentifier, path);
+        this.httpClient.delete(builtUrl, new HttpRequestConfig(), Void.class);
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application.
+     * @param path The path is implemented by the application
+     * @return application data
+     * @param request request object
+     * @see <ul>
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:84]:
```suggestion
     * @param applicationIdentifier identifier of the application
     * @param path The path is implemented by the application
     * @param request request object
     * @return application data
     * @see <ul>
```

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+     * <li><a href="https://developer.crowdin.com/api/v2/#operation/api.applications.api.put" target="_blank"><b>API Documentation</b></a></li>
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:85]: add "Enterprise API Documentation"

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+     * @param applicationIdentifier identifier of the application
+     * @param path The path is implemented by the application
+     * @return application data
+     * @param request request object
+     * @see <ul>
+     * <li><a href="https://developer.crowdin.com/api/v2/#operation/api.applications.api.put" target="_blank"><b>API Documentation</b></a></li>
+     * <li><a href="https://developer.crowdin.com/enterprise/api/v2/#operation/api.applications.api.put" target="_blank"><b>Enterprise API Documentation</b></a></li>
+     * </ul>
+     */
+    public ResponseObject<ApplicationData> updateOrReplaceApplicationData(String applicationIdentifier, String path, UpdateOrRestoreApplicationDataRequest request) throws HttpException, HttpBadRequestException {
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:47]: updateOr**Restore**ApplicationData

File: src/main/java/com/crowdin/client/applications/ApplicationsApi.java [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+    public ResponseObject<ApplicationData> getApplicationData(String applicationIdentifier, String path) throws HttpException, HttpBadRequestException {
+        String builtUrl = String.format("%s/applications/%s/api/%s", this.url, applicationIdentifier, path);
+        ApplicationDataResponseObject response = this.httpClient.get(builtUrl, new HttpRequestConfig(), ApplicationDataResponseObject.class);
+        return ResponseObject.of(response.getData());
+    }
+
+    /**
+     * @param applicationIdentifier identifier of the application
+     * @param path The path is implemented by the application
+     * @return application data
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/applications/ApplicationsApi.java:40]: put return after params

File: src/test/java/com/crowdin/client/applications/ApplicationsApiTest.java [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Test
+    void getApplicationData() {
+        ResponseObject<ApplicationData> response = this.getApplicationsApi().getApplicationData(applicationIdentifier, path);
+        assertNotNull(response);
+    }
+
+    @Test
+    void updateOrReplaceApplicationData() {
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/applications/ApplicationsApiTest.java:41]: updateOr**Restore**ApplicationData

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]

**Review by andrii-bodnar — APPROVED** [review]
@sugan0tech thank you!

**Pull request merged**